### PR TITLE
Add description to value relation widget

### PR DESCRIPTION
**Pull request by m-kuhn**
Add the possibility to add a description to each value in a value relation widget.
The description will show up as tooltip

Please excuse the different languages in the demo video.

Peek 2020-03-12 16-28

### CODE REVIEW

File: src/gui/editorwidgets/qgsvaluerelationwidgetwrapper.cpp [hunk 1]
```diff
@@ -365,6 +365,7 @@ void QgsValueRelationWidgetWrapper::populate( )
     for ( const QgsValueRelationFieldFormatter::ValueRelationItem &element : qgis::as_const( mCache ) )
     {
       whileBlocking( mComboBox )->addItem( element.value, element.key );
+      mComboBox->setItemData( mComboBox->count() - 1, element.description, Qt::ToolTipRole );
```
Comment (3nids) [inline src/gui/editorwidgets/qgsvaluerelationwidgetwrapper.cpp]: `element.description` might be an empty string. Better go with an invalid `QVariant` for the cases where no expression were given.

File: src/core/fieldformatter/qgsvaluerelationfieldformatter.h [hunk 2]
```diff
@@ -39,16 +39,18 @@ class CORE_EXPORT QgsValueRelationFieldFormatter : public QgsFieldFormatter
     struct ValueRelationItem
     {
       //! Constructor for ValueRelationItem
-      ValueRelationItem( const QVariant &key, const QString &value )
+      ValueRelationItem( const QVariant &key, const QString &value, const QString &description )
```
Comment (3nids) [inline src/core/fieldformatter/qgsvaluerelationfieldformatter.h]: I believe this is an API break
```suggestion
      ValueRelationItem( const QVariant &key, const QString &value, const QString &description = QString() )
```

**Review by 3nids — APPROVED** [review]

File: src/gui/editorwidgets/qgsvaluerelationwidgetwrapper.cpp (hunk 1 above)
Comment (m-kuhn) [reply src/gui/editorwidgets/qgsvaluerelationwidgetwrapper.cpp]:
```suggestion
      if ( !element.description.isEmpty() )
        mComboBox->setItemData( mComboBox->count() - 1, element.description, Qt::ToolTipRole );
```

**Comment by ponceta** [comment]
Thank you @m-kuhn @3nids !

**Comment by pigreco** [comment]
Please add `Widget` label

**Comment by m-kuhn** [comment]
@mbernasocchi can you add @pigreco to the qgis/community group so he can manage issues and labels?

**Comment by mbernasocchi** [comment]
> @mbernasocchi can you add @pigreco to the qgis/community group so he can manage issues and labels?

done

**Pull request merged**